pkg/platform/kube/platform.go

```diff
@@ -1260,3 +1272,47 @@ func (p *Platform) getDefaultServiceType() (v1.ServiceType, error) {
 		return "", errors.New("Not a valid configuration instance")
 	}
 }
+
+func (p *Platform) validateAPIGatewayFunctionsHaveNoIngresses(platformAPIGateway *platform.APIGatewayConfig) error {
+
+	// check ingresses on every upstream function
+	for _, upstream := range platformAPIGateway.Spec.Upstreams {
+		function, err := p.GetFunctions(&platform.GetFunctionsOptions{
+			Namespace: platformAPIGateway.Meta.Namespace,
+			Name: upstream.Nucliofunction.Name,
+		})
+		if err != nil {
+			return errors.New("Failed to get function")
+		}
+		if len(function) == 0 {
+
+			// if such function doesn't exist, just skip - because it doesn't have ingresses for sure
+			continue
+		}
+
+		ingresses := functionconfig.GetIngressesFromTriggers(function[0].GetConfig().Spec.Triggers)
+		if len(ingresses) > 0 {
+			return nuclio.NewErrConflict(fmt.Sprintf("Api gateway upstream function: %s must not have an ingress", upstream.Nucliofunction.Name))
+		}
+	}
+
+	return nil
+}
+
+// validate that a function is not exposed inside http triggers, while it is also exposed by an api gateway
```

Comment:
That is the key for this entire PR, mind you to put it on PR description as well?